### ISSUE TITLE
Error building polybar - make exits with error 2

### ISSUE THREAD

**Issue by Phantom-Rehan** [polybar/polybar#2182]
## Build Process
*Describe how you build polybar, list the exact commands you are using:*
```
mkdir build
cd build
cmake -DCMAKE_CXX_COMPILER="clang++"  -DPYTHON_XCBGEN=/usr/lib/python2.7/dist-packages ..
make -j$(nproc)
```
**If you use some other way (like the AUR):**

*List exactly where you are building from and which version you are building.*
Polybar 3.4.3 (tar) from the releases page

## Build Log
*Post everything that is output to the terminal while building polybar below. This HAS to include the output of the `cmake` and `make` commands, if you used them.*
```
mkdir build
cd build
cmake -DCMAKE_CXX_COMPILER="clang++"  -DPYTHON_XCBGEN=/usr/lib/python2.7/dist-packages ..
make -j$(nproc)
-- Trying to enable ccache
-- Couldn't locate ccache, disabling ccache...
-- The CXX compiler identification is Clang 10.0.0
-- Check for working CXX compiler: /usr/bin/clang++
-- Check for working CXX compiler: /usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Could not detect version with git, falling back to built-in version information.
-- No build type specified; using Release
-- Using supported compiler Clang-10.0.0
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Checking for module 'cairo-fc'
--   Found cairo-fc, version 1.16.0
-- Checking for module 'alsa'
--   Found alsa, version 1.2.2
-- Checking for module 'libcurl'
--   Found libcurl, version 7.68.0
-- Checking for module 'libmpdclient'
--   Found libmpdclient, version 2.18
-- Checking for module 'libnl-genl-3.0'
--   Found libnl-genl-3.0, version 3.4.0
-- Checking for module 'libpulse'
--   Found libpulse, version 13.99.1
-- Checking for module 'xcb-composite'
--   Found xcb-composite, version 1.14
-- Checking for module 'xcb-xkb'
--   Found xcb-xkb, version 1.14
-- Checking for module 'xcb-randr'
--   Found xcb-randr, version 1.14
-- Checking for module 'xcb-randr>=1.12'
--   Found xcb-randr, version 1.14
-- Checking for module 'xcb-xrm'
--   Found xcb-xrm, version 1.0
-- Checking for module 'xcb-cursor'
--   Found xcb-cursor, version 0.1.1
--  Build:
--    Version: 3.4.3
--    Type: Release
--    CXX: /usr/bin/clang++  -Wall -Wextra -Wpedantic -Wno-error=parentheses-equality -Wno-zero-length-array -O3 -DNDEBUG
--    LD: /usr/bin/ld  
--  Install Paths:
--    PREFIX:  /usr/local
--    BINDIR:  /usr/local/bin
--    DATADIR: /usr/local/share
--    DOCDIR:  /usr/local/share/doc/polybar
--    MANDIR:  /usr/local/share/man
--  Targets:
-- [X]   polybar-msg
-- [ ]   testsuite
-- [X]   documentation
--  Module support:
-- [X]   alsa (1.2.2)
-- [X]   curl (7.68.0)
-- [X]   i3
-- [X]   mpd (2.18)
-- [X]   network (libnl) (3.4.0)
-- [X]   pulseaudio (13.99.1)
-- [X]   xkeyboard (1.14)
--  X extensions:
-- [X]   xcb-randr (1.14)
-- [X]   xcb-randr (monitor support) (1.14)
-- [X]   xcb-composite (1.14)
-- [X]   xcb-xkb (1.14)
-- [X]   xcb-xrm (1.0)
-- [X]   xcb-cursor (0.1.1)
--  Log options:
-- [X]   Trace logging
-- The C compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Checking for module 'xcb-proto'
--   Found xcb-proto, version 1.14
-- Found PythonInterp: /home/styx/.platformio/penv/bin/python3 (found suitable version "3.8.2", minimum required is "3.5") 
-- XCB[XCB]: Found component XCB
-- Found XCB_XCB: /usr/lib/x86_64-linux-gnu/libxcb.so  
-- XCB[ICCCM]: Found component ICCCM
-- Found XCB_ICCCM: /usr/lib/x86_64-linux-gnu/libxcb-icccm.so  
-- XCB[EWMH]: Found component EWMH
-- Found XCB_EWMH: /usr/lib/x86_64-linux-gnu/libxcb-ewmh.so  
-- XCB[UTIL]: Found component UTIL
-- Found XCB_UTIL: /usr/lib/x86_64-linux-gnu/libxcb-util.so  
-- XCB[IMAGE]: Found component IMAGE
-- Found XCB_IMAGE: /usr/lib/x86_64-linux-gnu/libxcb-image.so  
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so;/usr/lib/x86_64-linux-gnu/libxcb-icccm.so;/usr/lib/x86_64-linux-gnu/libxcb-ewmh.so;/usr/lib/x86_64-linux-gnu/libxcb-util.so;/usr/lib/x86_64-linux-gnu/libxcb-image.so  
-- Searching for xcbgen with python
-- Searching for xcbgen with python3
-- Searching for xcbgen with python2
-- Searching for xcbgen with python2.7
-- XCB[RANDR]: Found component RANDR
-- Found XCB_RANDR: /usr/lib/x86_64-linux-gnu/libxcb-randr.so  
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so;/usr/lib/x86_64-linux-gnu/libxcb-icccm.so;/usr/lib/x86_64-linux-gnu/libxcb-ewmh.so;/usr/lib/x86_64-linux-gnu/libxcb-util.so;/usr/lib/x86_64-linux-gnu/libxcb-image.so;/usr/lib/x86_64-linux-gnu/libxcb-randr.so  
-- XCB[COMPOSITE]: Found component COMPOSITE
-- Found XCB_COMPOSITE: /usr/lib/x86_64-linux-gnu/libxcb-composite.so  
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so;/usr/lib/x86_64-linux-gnu/libxcb-icccm.so;/usr/lib/x86_64-linux-gnu/libxcb-ewmh.so;/usr/lib/x86_64-linux-gnu/libxcb-util.so;/usr/lib/x86_64-linux-gnu/libxcb-image.so;/usr/lib/x86_64-linux-gnu/libxcb-randr.so;/usr/lib/x86_64-linux-gnu/libxcb-composite.so  
-- XCB[XKB]: Found component XKB
-- Found XCB_XKB: /usr/lib/x86_64-linux-gnu/libxcb-xkb.so  
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so;/usr/lib/x86_64-linux-gnu/libxcb-icccm.so;/usr/lib/x86_64-linux-gnu/libxcb-ewmh.so;/usr/lib/x86_64-linux-gnu/libxcb-util.so;/usr/lib/x86_64-linux-gnu/libxcb-image.so;/usr/lib/x86_64-linux-gnu/libxcb-randr.so;/usr/lib/x86_64-linux-gnu/libxcb-composite.so;/usr/lib/x86_64-linux-gnu/libxcb-xkb.so  
-- xpp: including xcb proto composite.xml
-- xpp: including xcb proto randr.xml
-- xpp: including xcb proto xkb.xml
-- xpp: including xcb proto xproto.xml
-- Checking for module 'jsoncpp>=1.7.7'
--   Requested 'jsoncpp >= 1.7.7' but version of jsoncpp is 1.7.4
You may find new versions of jsoncpp at https://github.com/open-source-parsers/jsoncpp
-- Could not find jsoncpp>=1.7.7 -- building static lib...
-- JsonCpp Version: 1.7.7
-- Found PythonInterp: /home/styx/.platformio/penv/bin/python3 (found suitable version "3.8.2", minimum required is "2.6") 
-- Font not found: fixed:pixelsize=10
-- Font not found: unifont:fontformat=truetype
-- Font not found: siji:pixelsize=10
-- Configuring done
-- Generating done
-- Build files have been written to: /home/phantom/Downloads/polybar/build
Scanning dependencies of target doc_html
Scanning dependencies of target doc_man
Scanning dependencies of target jsoncpp_lib_static
[  1%] Generating ../../../lib/xpp/include/xpp/proto/x.hpp
[  1%] sphinx-build html: see doc/builder-html.log
[  2%] sphinx-build man: see doc/builder-man.log
[  3%] Building CXX object lib/i3ipcpp/libs/jsoncpp-1.7.7/src/lib_json/CMakeFiles/jsoncpp_lib_static.dir/json_reader.cpp.o
Traceback (most recent call last):
  File "/home/phantom/Downloads/polybar/lib/xpp/generators/cpp_client.py", line 3101, in <module>
    from xcbgen.state import Module
ModuleNotFoundError: No module named 'xcbgen'
make[2]: *** [lib/xpp/CMakeFiles/xpp.dir/build.make:74: ../lib/xpp/include/xpp/proto/x.hpp] Error 1
make[2]: *** Deleting file '../lib/xpp/include/xpp/proto/x.hpp'
make[1]: *** [CMakeFiles/Makefile2:641: lib/xpp/CMakeFiles/xpp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  4%] Building CXX object lib/i3ipcpp/libs/jsoncpp-1.7.7/src/lib_json/CMakeFiles/jsoncpp_lib_static.dir/json_value.cpp.o
[  4%] Built target doc_man
[  5%] Building CXX object lib/i3ipcpp/libs/jsoncpp-1.7.7/src/lib_json/CMakeFiles/jsoncpp_lib_static.dir/json_writer.cpp.o
/home/phantom/Downloads/polybar/lib/i3ipcpp/libs/jsoncpp-1.7.7/src/lib_json/json_value.cpp:73:15: error: implicit conversion from 'long' to 'double' may lose precision [-Werror,-Wimplicit-int-float-conversion]
  return d >= min && d <= max;
           ~~ ^~~
/home/phantom/Downloads/polybar/lib/i3ipcpp/libs/jsoncpp-1.7.7/src/lib_json/json_value.cpp:762:25: note: in instantiation of function template specialization 'Json::InRange<long, long>' requested here
    JSON_ASSERT_MESSAGE(InRange(value_.real_, minInt64, maxInt64),
                        ^
/home/phantom/Downloads/polybar/lib/i3ipcpp/libs/jsoncpp-1.7.7/src/lib_json/json_value.cpp:73:27: error: implicit conversion from 'long' to 'double' may lose precision [-Werror,-Wimplicit-int-float-conversion]
  return d >= min && d <= max;
                       ~~ ^~~
/home/phantom/Downloads/polybar/lib/i3ipcpp/libs/jsoncpp-1.7.7/src/lib_json/json_value.cpp:73:27: error: implicit conversion from 'unsigned long' to 'double' may lose precision [-Werror,-Wimplicit-int-float-conversion]
  return d >= min && d <= max;
                       ~~ ^~~
/home/phantom/Downloads/polybar/lib/i3ipcpp/libs/jsoncpp-1.7.7/src/lib_json/json_value.cpp:783:25: note: in instantiation of function template specialization 'Json::InRange<int, unsigned long>' requested here
    JSON_ASSERT_MESSAGE(InRange(value_.real_, 0, maxUInt64),
                        ^
WARNING: html_static_path entry '/home/phantom/Downloads/polybar/doc/_static' does not exist
3 errors generated.
make[2]: *** [lib/i3ipcpp/libs/jsoncpp-1.7.7/src/lib_json/CMakeFiles/jsoncpp_lib_static.dir/build.make:76: lib/i3ipcpp/libs/jsoncpp-1.7.7/src/lib_json/CMakeFiles/jsoncpp_lib_static.dir/json_value.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[  5%] Built target doc_html
make[1]: *** [CMakeFiles/Makefile2:696: lib/i3ipcpp/libs/jsoncpp-1.7.7/src/lib_json/CMakeFiles/jsoncpp_lib_static.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

```

## Environment:
Ubuntu 20.04

## Additional context
*Add any other context that you think is necessary about the problem here.*
I had the same problem as this -
https://github.com/polybar/polybar/issues/2031
So i added the 
-DPYTHON_XCBGEN=/usr/lib/python2.7/dist-packages to the cmake command
Also, for whatever reason it might be useful, my home directory is not the same as /home/username

**Comment by Phantom-Rehan** [comment]
My issue has not been fixed, but since, i was able to install polybar using a .deb file for ubuntu from [here](https://pkgs.org/download/polybar) It has to be noted, however, that the deb file was for **GROOVY** and **NOT FOCAL FOSSA **, which is the ubuntu distro i am using

**Comment by patrick96** [comment]
Well, looks like building 3.4.3 on ubuntu 20.04 is broken now...

The problem is that up until 3.4.3 we manually compile jsoncpp version if it is not available or below version 1.7.7 (which is the case on Ubuntu), this compilation fails now (probably because of a new compiler version).

This is already fixed by #2015, just not part of a release yet.

As a workaround either compile against the `master` branch or edit the file `lib/i3ipcpp/CMakeLists.txt`, change the line `pkg_check_modules(JSONCPP jsoncpp>=1.7.7)` to `pkg_check_modules(JSONCPP jsoncpp)`, and build again.
